### PR TITLE
add to_global_netpol_calico() for baseline rules without namespaces

### PR DESCRIPTION
**Pull request by adisos**

### CODE REVIEW

File: src/selector.py [hunk 1]
```diff
@@ -70,15 +79,18 @@ class LabelSelector(Selector):
...
         """
         :param str key: Label key
         :param SelectorOp op: The selector operator
         :param list values: A list of label values
+        :param str calico_selector_expr: the Selector expr by Calico syntax
         """
         self.key = key
         self.operator = op
         self.values = values
+        self.calico_selector_expr = calico_selector_expr
```
Comment (zivnevo) [inline src/selector.py]: Better produce the calico-style selector on-the-fly

File: src/baseline_rule.py [hunk 2]
```diff
@@ -128,15 +145,102 @@ def get_port_array(self):
...
+            'spec': policy_spec
+        }
+
+    def to_global_netpol_calico(self):
+        """
+        Create Calico GlobalNetworkPolicy resources representing the connections specified by the rule.
+        Relevant for baseline rules with no namespace, thus using GlobalNetworkPolicy which applies to all namespaces.
+        Note that two GlobalNetworkPolicy resources may be required for allowing both directions of a connection.
+
+        :return: A Calico GlobalNetworkPolicy resources representing the connections specified by the rule
```
Comment (zivnevo) [inline src/baseline_rule.py:196]:
```suggestion
        :return: One or two Calico GlobalNetworkPolicy resources representing the connections specified by the rule
```

File: src/baseline_rule.py [hunk 3]
```diff
@@ -128,15 +145,102 @@ def get_port_array(self):
...
+        :rtype: (dict, union[dict, None])
+        """
+        is_ingress_policy, policy_type = self._get_policy_type()
+        policy_spec = dict()
+        policy_spec.update({'types': [policy_type]})
```
Comment (zivnevo) [inline src/baseline_rule.py:201]:
```suggestion
        policy_spec = {'types': [policy_type]}
```

File: src/baseline_rule.py [hunk 4]
```diff
@@ -128,15 +145,102 @@ def get_port_array(self):
...
+        policy_spec.update(policy_selector)
+        ports_list, protocol = self.get_port_array_calico()
+        ports_dict = {'ports': ports_list}
+
+        rule_to_add = {'action': 'Allow'}
+        if protocol:
+            rule_to_add.update({'protocol': protocol})
+        if is_ingress_policy:
+            src_dict = self._selectors_as_netpol_peer_calico(self.source, False)
+            rule_to_add.update({'source': src_dict})
```
Comment (zivnevo) [inline src/baseline_rule.py:213]:
```suggestion
            rule_to_add['source'] = src_dict
```
Avoid using `dict.update()` (to be more consistent with the rest of the code)

File: src/baseline_rule.py [hunk 5]
```diff
@@ -128,15 +145,102 @@ def get_port_array(self):
 
         return ports_array
 
+    def get_port_array_calico(self):
+        if not self.port_min:
+            return [], self.protocol
+        ports_array = []
+        for port in range(self.port_min, self.port_max + 1):
+            ports_array.append(port)
+
```
Comment (zivnevo) [inline src/baseline_rule.py:154]: Calico and K8s have support for port ranges - should we use it? It could be more efficient

**Review by zivnevo — CHANGES_REQUESTED** [review]

File: src/selector.py (hunk 1 above)
Comment (adisos) [reply src/selector.py]: done

File: src/baseline_rule.py (hunk 2 above)
Comment (adisos) [reply src/baseline_rule.py]: done

File: src/baseline_rule.py (hunk 3 above)
Comment (adisos) [reply src/baseline_rule.py]: done

File: src/baseline_rule.py (hunk 4 above)
Comment (adisos) [reply src/baseline_rule.py]: done

File: src/baseline_rule.py (hunk 5 above)
Comment (adisos) [reply src/baseline_rule.py]: done

File: src/baseline_rule.py [hunk 6]
```diff
@@ -121,22 +138,115 @@ def get_port_array(self):
         if not self.port_min:
             return []
         ports_array = []
-        for port in range(self.port_min, self.port_max + 1):
-            port_rec = {'protocol': self.protocol} if self.protocol else {}
-            port_rec['port'] = port
-            ports_array.append(port_rec)
+        port_rec = {'protocol': self.protocol} if self.protocol else {}
```
Comment (zivnevo) [inline src/baseline_rule.py:141]: The code (also the old code) doesn't properly handle the case where `protocol` is set, but `port_min` is not set.
I think the code for Calico works well.

**Review by zivnevo — CHANGES_REQUESTED** [review]
Need one more fix for an old bug

File: src/baseline_rule.py [hunk 7]
```diff
@@ -121,22 +138,115 @@ def get_port_array(self):
...
+        }
+
+    def to_global_netpol_calico(self):
+        """
+        Create Calico GlobalNetworkPolicy resources representing the connections specified by the rule.
+        Relevant for baseline rules with no namespace, thus using GlobalNetworkPolicy which applies to all namespaces.
+        Note that two GlobalNetworkPolicy resources may be required for allowing both directions of a connection.
+
+        :return: One or two Calico GlobalNetworkPolicy resources representing the connections specified by the rule
+        :rtype: (dict, union[dict, None])
```
Comment (zivnevo) [inline src/baseline_rule.py:204]: After looking at the code in baseline-rules-verifier, it seems like its better if this function returns a list

**Review by zivnevo — CHANGES_REQUESTED** [review]
One more small change

File: src/baseline_rule.py (hunk 6 above)
Comment (adisos) [reply src/baseline_rule.py:138]: done

File: src/baseline_rule.py (hunk 7 above)
Comment (adisos) [reply src/baseline_rule.py]: done

**Review by zivnevo — APPROVED** [review]

**Pull request merged**